source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -106,6 +114,34 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   return Http::FilterHeadersStatus::StopIteration;
 }
 
+bool Filter::requestAllowed(absl::Span<const RateLimit::LocalDescriptor> request_descriptors) {
+  return getRateLimiter()->requestAllowed(request_descriptors);
+}
+
+LocalRateLimiterImplSharedPtr Filter::getRateLimiter() {
+  const auto* config = getConfig();
+
+  if (!decoder_callbacks_->streamInfo().filterState()->hasData<PerConnectionRateLimiter>(
+          PerConnectionRateLimiter::key())) {
+    auto rate_limiter = std::make_shared<Filters::Common::LocalRateLimit::LocalRateLimiterImpl>(
+        std::chrono::milliseconds(
+            PROTOBUF_GET_MS_OR_DEFAULT(config->protoConfig().token_bucket(), fill_interval, 0)),
```

Comment:
we can avoid doing this `std::chrono::milliseconds()` creation on every new connection by doing it once when constructing the FilterConfig object.

Reply:
This makes sense. I've made the appropriate amends 👍